### ISSUE TITLE
Duplicati stalls during "verifying backend data"

### ISSUE THREAD

**Issue by StevenKSanford** [duplicati/duplicati#2335]
I have:
- [x ] searched open and closed issues for duplicates

----------------------------------------

### Version info

**Duplicati Version:** 2.0.1.48_canary_2017-02-23
**Operating System:** MACOS Sierra 10.2.3
**Backend:** Google Drive

### Bug description
I have repeatedly, on two of my four backup sets, had Duplicati stall in the "verifying backend data" step after a few minutes of processing a backup or repair request, then mono uses 100% of one CPU until I terminate it. The other two backup sets run fine. Both of these at 2-8 times larger than the ones that fail.

The last entry in the log is always something like:

Starting - ExecuteReader: SELECT DISTINCT "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Hash", "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Size", "File"."Path", "BlocksetEntry"."Index" * 102400 FROM "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B", "Block", "BlocksetEntry", "File" WHERE "File"."BlocksetID" = "BlocksetEntry"."BlocksetID" AND "Block"."ID" = "BlocksetEntry"."BlockID" AND "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Hash" = "Block"."Hash" AND "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Size" = "Block"."Size" AND "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Restored" = ?

I have tried repairing and delete-repairing the affected databases, but Duplicati still stalls on the next backup.

### Steps to reproduce

- run backup to Google Drive

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)

- Duplicati stalls in the "verifying backend data" phase, with "mono-sgen32" using 100% of a CPU.

The last entry in the log is always something like:

Starting - ExecuteReader: SELECT DISTINCT "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Hash", "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Size", "File"."Path", "BlocksetEntry"."Index" * 102400 FROM "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B", "Block", "BlocksetEntry", "File" WHERE "File"."BlocksetID" = "BlocksetEntry"."BlocksetID" AND "Block"."ID" = "BlocksetEntry"."BlockID" AND "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Hash" = "Block"."Hash" AND "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Size" = "Block"."Size" AND "MissingBlocks-8DD87D3AED9A00449556CB5217E3974B"."Restored" = ?

This will run like this for days with no further log messages. 

Quitting "mono-sgen32" from the Activity Monitor (or rebooting) and restarting Duplicati is the only way to resume access to Duplicati. When Duplicati is restarted, it has exited the backup that stalled, but leaves no further log messages (see above).

Manage Database - repair will stall at the same point. Database delete/repair completes successfully, but the next backup still stalls. Attempts to create an bug report also stall

**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)

- backup should run to completion

### Screenshots
see log entry above

### debug log
attempts to create an bug report also stall

**Comment by StevenKSanford** [comment]
Update: another backup dataset has stalled in the "verifying backend data" step. 

I just completed a successful backup a few days ago on this fileset, but received a warning that a file was missing from the backup, and that I should run "repair". I ran the repair process and it completed successfully. The next backup attempt on this fileset stalled. 

I believe that this has been the sequence for each stalled fileset.

**Comment by afalout** [comment]
Almost the came here;

2.0.1.53_experimental_2017-03-13 Debian 64 bit with Amazon Drive back-end
Netgear ReadyNad RC316 with 4GB RAM and quad core Intel CPU

Had everything working fine for few months until last week

Whatever I do, it goes into "Verifying backend data" and just sits there. Zero activity, for days. Have 4 backup jobs configured and all behave the same.

Live log display with profiling on stops at the same point:


    Apr 14, 2017 1:38 PM: Starting - ExecuteScalar: SELECT COUNT (*) FROM (SELECT DISTINCT "Block"."Hash", "Block"."Size" FROM "Block", "RemoteVolume" WHERE "RemoteVolume"."ID" = "Block"."VolumeID" AND "RemoteVolume"."State" NOT IN (?,?,?,?))
    Apr 14, 2017 1:38 PM: ExecuteReader: SELECT "Name" FROM "RemoteVolume" WHERE "Type" = ? AND NOT "ID" IN (SELECT "BlockVolumeID" FROM "IndexBlockLink") AND "State" IN (?,?) took 00:00:00.047
    Apr 14, 2017 1:38 PM: Starting - ExecuteReader: SELECT "Name" FROM "RemoteVolume" WHERE "Type" = ? AND NOT "ID" IN (SELECT "BlockVolumeID" FROM "IndexBlockLink") AND "State" IN (?,?)
    Apr 14, 2017 1:38 PM: ExecuteReader: SELECT "ID", "Timestamp" FROM "Fileset" ORDER BY "Timestamp" DESC took 00:00:00.000
    Apr 14, 2017 1:38 PM: Starting - ExecuteReader: SELECT "ID", "Timestamp" FROM "Fileset" ORDER BY "Timestamp" DESC
    Apr 14, 2017 1:38 PM: PreBackupVerify took 00:01:01.890
    ...et...


Another example:


Apr 14, 2017 1:05 PM: Starting - ExecuteScalar: SELECT COUNT (*) FROM (SELECT DISTINCT "Block"."Hash", "Block"."Size" FROM "Block", "RemoteVolume" WHERE "RemoteVolume"."ID" = "Block"."VolumeID" AND "RemoteVolume"."State" NOT IN (?,?,?,?))
Apr 14, 2017 1:05 PM: ExecuteReader: SELECT "Name" FROM "RemoteVolume" WHERE "Type" = ? AND NOT "ID" IN (SELECT "BlockVolumeID" FROM "IndexBlockLink") AND "State" IN (?,?) took 00:00:00.029
Apr 14, 2017 1:05 PM: Starting - ExecuteReader: SELECT "Name" FROM "RemoteVolume" WHERE "Type" = ? AND NOT "ID" IN (SELECT "BlockVolumeID" FROM "IndexBlockLink") AND "State" IN (?,?)
Apr 14, 2017 1:05 PM: ExecuteReader: SELECT "ID", "Timestamp" FROM "Fileset" ORDER BY "Timestamp" DESC took 00:00:00.000
Apr 14, 2017 1:05 PM: Starting - ExecuteReader: SELECT "ID", "Timestamp" FROM "Fileset" ORDER BY "Timestamp" DESC
Apr 14, 2017 1:05 PM: PreBackupVerify took 00:01:02.752
...etc


Googled for few hours but found no applicable solutions - some talking about 1.3.x versions, some with unrelated causes.


I also noted the still open ticket on Github: https://github.com/duplicati/duplicati/issues/1735

Also posted on Google group, but there was no response.

Can anyone suggest what to do ?


Any help very much appreciated

**Comment by StevenKSanford** [comment]
afalout: I am experimenting with a workaround for this problem: 

NOTE: BE SURE THAT YOU HAVE AN ALTERNATE BACKUP COPY FOR ALL OF YOUR FILES - DUPLICATI SHOULD NOT BE YOUR ONLY BACKUP.

(1) For any backups that have stalled, remove (delete or move) only the files that have been created since the last successful backup (down to but NOT including the last successful ...dlist.zip.aes file) ON THE REMOTE target drive (Google Drive, Amazon Drive, etc.). You should be able to sort your remote target drive by file-created date to see what's been created since the last successful backup. Do NOT delete files on your local source drive. Give the remote drive some time to process the delete or move requests if there are a lot of files involved.

(2) Delete and recreate your local Duplicati backup database(s) for the stalled backups

(3) Trying running Duplicati backup again. You may get some "missing files" or "duplicate files" errors, which should not be fatal, when recreating the database(s) or running the backup(s).

This may not prevent the problem from re-occurring sometime in the future, but at least it may allow you to resume backing up your files. 

Good luck... Steve

**Comment by afalout** [comment]
Hi Steve, thanks for your help;

1) The backup of my primary interest (my music files, 12T) never completed successfully; it was running (successfully, apparently) for a while, then at about 20% it started getting killed repeatedly, running out of RAM on my 2GB Netgear NAS (which is as I understand it, another issue affecting number of users, eg. unexpected RAM utilization). Since then I upgraded to 4GB RAM as I could not find the other solution. After that it started hanging immediately as described, on "verifying backed data...". (The two may or may not be related, but FYI). Therefore, there are no files that where created "since the last successful backup" on this backup. (I will try this on other, smaller backups)

2) Is there a point it trying to delete and recreated local DB, considering what I explained in 1) above? If so, please see my note #5 below...

3) Is there anything I can do to help you fix this to "prevent the problem from re-occurring sometime in the future"? Seems this behaviour affects a significant number of users over significant number of versions going back at least 2 years?

4) Or did I misunderstand - was the root cause of this issue resolved somehow? If so, in which version? If not, I wonder why was the bug report closed?

5) I have two identical, synced NAS boxes (using Syncthing). I exported Duplicati backup definitions from the first (stuck) NAS, and imported them into the second NAS/Duplicati. I then initiated "Recreate" maintenance task on the second NAS - primarily out of curiosity, and possibly hoping that maybe I can continue backing up data from the second NAS, in case I cant fix the first one (which might be a dumb idea) The reason why I'm mentioning this, is that this "recreating database...." has been running for last 2 weeks, and it reached maybe 50% (although with steady progress indicator movement). Considering that this is re-creating a database of only about 20% finished backup (of 12T), unless I'm doing something wrong, this would mean that the process is not viable for practical use. Eg, it takes about same or comparable time to backup this amount of data, as it takes to recreate the database for the same data? (NB. I have 900 Mbps fibre Internet symmetrical up/down)

Your help is very much appreciated.
Andrej

**Comment by StevenKSanford** [comment]
Andrej:

Thanks for your update and additional information. I am the one who reported this problem, not a programmer with the Duplicati team, so my suggestions are based solely on my experience as a Duplicati user. Sorry if this wasn't' clear.

To answer your point #2, I have had mixed success with deleting and recreating the database. Some of my backups have begun working again, while one is still hung (so far). I'm waiting on a different (multi-day) backup to complete so I can test this on the remaining hung backup. The backup that is running now was previously hung, so some combination of deleting/recreating the database and deleting target drive files may have worked, or the problem resolved itself somehow with incremental revisions to Duplicati (canary).

Good luck with your other issues. I'm not qualified to comment on them nor to make suggestions.

...Steve

**Comment by StevenKSanford** [comment]
I am far from concluding that this issue is resolved, so I am leaving it open for now.

**Comment by StevenKSanford** [comment]
I seem to be on the right track to solving this problem, but the solution is a little severe: go to the remote repository and delete all backup files back to the end of the PREVIOUS successful backup BEFORE THE LAST successful backup. You should keep the "...dlist.zip.aes" from that PREVIOUS backup. I know that this means that you're deleting the files from a backup that Duplicati thinks is successful, but because you haven't been able to run a successful backup since the last successful backup, odd are good that the last successful backup is damaged.

Still testing... Steve

**Comment by StevenKSanford** [comment]
Update.... the above strategy appears to be working, that is, delete everything on the remote backup back to (but not including) the ...dlist... file of the backup BEFORE THE LAST SUCCESSFUL BACKUP (using the "last successful backup" date on the GUI home screen), delete/rebuild the database, run backup. So far I've gotten two of four stalled backups to run without stalling on the "verifying backend" step.

One caveat... if you have to reboot your machine during a backup, you have to start all over. Duplicati does not seem to be able to recover from sudden termination. The "failed, attempting cleanup" message in the log is, apparently, useless in terms of fixing database problems.

Note also that it rarely "stops" when I click the "stop backup" button (no matter which type of "stop" I choose).... it just ignores the "stop backup" request. My backups tend to take several days (multi-TB source drives), even if there's very little new on the source drive, so the risk of reboot is fairly high.

One more caution: I've never tried to recover a Duplicati backed-up file, so I don't know if my backups are any good.

Has anyone else had success trying this method to get past the stalled backup problem?

Thanks... Steve

**Comment by StevenKSanford** [comment]
The above strategy seems to have worked. All of my Duplicati backups are running again. Closing this report. Thanks... Steve

**Comment by rkarlsba** [comment]
Seems my duplicati hung last night and I gave the VM a reboot. Now it's been sitting at Verifying backend data ... for several hours. Any idea what I can do about this? I have backed up 1,5TiB or so already, so I don't really want to start all over. Not for the bandwidth, I currently have 300Mbps symmetric, but I can't use the whole of that for some reason. Backend is jotta.

Thanks

roy

**Comment by mhellmeier** [comment]
Same problem here like  @rkarlsba : 
Duplicati hangs up last night, I rebooted my maschine. Now it hangs on "Verifying backend data...".
I backup to Amazon Cloud Drive.

**Comment by CrazyDymond88** [comment]
I am still actively experiencing this issue int he latest version of the Duplicati client.  The workaround works but it's not a solution if you can't reboot a machine and resume backup after booting.  Please reopen this issue, it is an open issue.

**Comment by StevenKSanford** [comment]
Sorry you still have this problem.... mine has not reoccurred since I followed my "delete all back to last backup list" protocol on all of my failing backups. I frequently reboot now without mishap. Most of the time, I can pause Duplicati before rebooting. I can no longer reproduce this problem, but I will reopen the problem so that other can pursue getting this fixed.... Steve

**Comment by CrazyDymond88** [comment]
Interesting, in my case this happened during the initial backup, on both backup to Google Drive and to a smb file share. I've deleted all backup files and databases and restarted.  Good point to pause before a reboot, I will monitor the situation over the next few days and report back here.  Thanks!

**Comment by CrazyDymond88** [comment]
After pausing the backup and rebooting I now have a "completing previous backup" message with no way of knowing whether it is actually doing anything.  I don't see much disk or network activity so I'm assuming it's not running correctly, but as the backup was a couple of Tb I'll give it a day or so and see what happens.

**Comment by mhellmeier** [comment]
After rebooting the maschine without pausing the backup, it stocks on "verifying backend data". Your workaround, @StevenKSanford , works. But I can reproduce this problem.
If your pc crashes (e.g. due to a power failure), duplicati hangs: It is very anoying, deleting the last broken files and recreating the local databaste every time your pc fails.

**Comment by get2him** [comment]
I'm experiencing the same problem.
Stuck with "Verifying backend data ... " 
It starts with one of the backup jobs and then spreads to all other backup jobs setup..
The described workaround does not work for me. The task shows no sign of life. 
The log (live:profiling) does not show any errors.

What can I do?
Last time I had to delete all backup files and recreate everything. But to do that every month just contradicts the idea of an backup...

Does it help to use the 1.xx version of duplicati?

**Comment by CrazyDymond88** [comment]
I have abandoned the product altogether given the myriad of issues I've had trying to complete my first backups, including this one.  It's sad, this product has such great potential.

**Comment by get2him** [comment]
I deleted my previous post because, I figured out what the problem was.
My FTP server IP address has changed. 
The fun part was to figure this out, because duplicati didn't posted any error message indicating this problem to the log. Instead it told me ""verifying backend data".
The developers must have a sadomachistic disposition :)

**Comment by NoLooseEnds** [comment]
@CrazyDymond88 
That's the reason why it is important to try to report bugs, and hopefully be of help to get it fixed. It's free after all (or you can donate).

@get2him
Yes, that is one of the things that should be improved. Better information about what exactly the Duplicati is doing, and where/why it fails.

**Comment by instantlinux** [comment]
This bug is 100% reproducible under 2.0.2.1-beta, and it's one that I'd put on a Top-10 list of things that need fixing (SOON!) because of how it undermines confidence in the entire system.

Here's how to reproduce it (I've done it quite a few times):
- Set up Docker Engine (I'm running a swarm of 3 nodes under 17.09-ce)
- Set up a NAS drive to which you want your data transferred; mount to your docker host using NFS
- Run a duplicati container such as linuxserver/duplicati:latest
- Configure a local Duplicati backup with the destination path name of your NFS mount

I was never able to transfer more than about 150-200GB of data that way, the cluster would go unstable and force restart of duplicati before completing my 400GB backup. But you can use this scenario to reproduce this at will without waiting for 200GB to transfer: simply invoke "docker restart" to interrupt duplicati while it's running a backup. Almost every time, Duplicati's internal database gets corrupted and you have to delete/regenerate your entire save set.

**Comment by afalout** [comment]
Just in case it helps anyone from wasting weeks of there time, I will share my observation of this, and many other Duplicati issues.... In my opinion, as I have no other.

**This software is not ready for any practical use.** Issues such as this one, are something developers should sort out BEFORE they ask anyone to even look at there work.

It may look promising, but it's broken in several fatal ways. Its been almost a year, and this issue is still present. 

While I do appreciate the developers effort, and there willingness to share that effort, I do not appreciate what amounts to false advertising. For free or not, this project wasted many, many, many frustrating hours and days of many, many people - by **misleading them into thinking that this software is in functional state, and can be productively used by typical backup software users.**

Only constructive recommendation I can make at this time, is to add a big red title to the top of the home page, that reads something like "WARNING - THIS IS ALPHA QUALITY SOFTWARE AND IS NOT READY FOR USE BY ANYONE OTHER THEN DEVELOPERS THAT WANT TO CONTRIBUTE TO MAKING IT BETTER".

All the best

**Comment by kimmerin** [comment]
After losing five backups in the last year due to this bug I can only agree to the previous comments about the quality of this software. But I try to be constructive...

With the new canary version you can see a bit more in the status view of the backup what it's currently doing so I can give some infos about it. The situation here is the following: Every time a running backup is interrupted (e.g. the system is shut down before the backup is finished), the whole backup is broken. the next time the backup starts it's starting a "Verifying backend data" process and seems to land in some kind of endless loop while reading in sqlite database. I created a dump using Process Monitor and you can see that it's reading the same part of the database over and over again (with ca. 300 MB/s). Funnily enough the status information in the browser shows a file as current file that was created later than the last backup took place, so I'm not sure what's it looking for in the database.

The database itself BTW was deleted and newly created.

In addition to the ProcMon-dump I looked into the threads of the process using Process Explorer. There is one thread with the start address clr.dll!GetMetaDataInternalInterfaceFromPublic+0x61090 that takes most of the cicles of the single CPU the process is using. I copied some stacks of it that you can find at the end of this message. I try to upload the ZIP containing the dump in a separate message (27 MB that will lead to 400 MB uncompressed data)

Hope that helps to track this bug down that simply prevents me from having a working backup most of the time. BTW: I installed the canary because the stable versions down to version 1 have the same issue, so the only alternative really would be to put some considerable distance between yourself and Duplicati.


ntdll.dll!ZwReadFile+0xa
KERNELBASE.dll!LocalReAlloc+0x131
KERNEL32.dll!ReadFile+0x59
SQLite.Interop.dll!sqlite3_os_init+0x3472
SQLite.Interop.dll!sqlite3_enable_shared_cache+0x84b1
SQLite.Interop.dll!sqlite3_enable_shared_cache+0x5bfb
SQLite.Interop.dll!sqlite3_backup_init+0xb561
SQLite.Interop.dll!sqlite3_backup_init+0x7f4b
SQLite.Interop.dll!sqlite3_backup_init+0x78db
SQLite.Interop.dll!sqlite3_expired+0x1c95
SQLite.Interop.dll!sqlite3_blob_open+0x270f
SQLite.Interop.dll!sqlite3_step+0x2fd
SQLite.Interop.dll!sqlite3_step+0x9c
[Native Frame: IL Method without Metadata]
[Managed to Unmanaged Transition]
System.Data.SQLite.dll!System.Data.SQLite.SQLite3.Step+0x2f3
System.Data.SQLite.dll!System.Data.SQLite.SQLiteDataReader.NextResult+0x1b7
System.Data.SQLite.dll!System.Data.SQLite.SQLiteDataReader..ctor+0x13c
System.Data.SQLite.dll!System.Data.SQLite.SQLiteCommand.ExecuteReader+0x3c
Duplicati.Library.Main.dll!Duplicati.Library.Main.Database.ExtensionMethods.ExecuteReader+0x17b
Duplicati.Library.Main.dll!<GetBlocklists>d__88.MoveNext+0x1a5
Duplicati.Library.Main.dll!<CreateIndexVolume>d__0.MoveNext+0x5f2
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!MoveNextRunner.Run+0x6c
mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction+0x61
mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations+0x3da
mscorlib.dll!System.Threading.Tasks.Task`1.TrySetResult+0x65
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.TrySetResult+0x14
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.SetResult+0xa
Duplicati.Library.Main.dll!<<DoRunOnMain>b__1>d.MoveNext+0x18e
mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start+0xb1
Duplicati.Library.Main.dll!<>c__DisplayClass5_0`1.<DoRunOnMain>b__1+0x93
Duplicati.Library.Main.dll!<Start>d__4.MoveNext+0x118
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!MoveNextRunner.Run+0x6c
mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction+0x61
mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations+0x3da
mscorlib.dll!System.Threading.Tasks.Task`1.TrySetResult+0x65
mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetResult+0x9a
CoCoL.dll!<ReadAsync>d__29.MoveNext+0xc95
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!MoveNextRunner.Run+0x6c
mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction+0x61
mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations+0x3da
mscorlib.dll!System.Threading.Tasks.Task`1.TrySetResult+0x65
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.TrySetResult+0x14
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.SetResult+0xa
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem+0x70
mscorlib.dll!System.Threading.ThreadPoolWorkQueue.Dispatch+0x14e
[Unmanaged to Managed Transition]
clr.dll!LogHelp_LogAssert+0x1013
clr.dll!LogHelp_LogAssert+0xeb6
clr.dll!LogHelp_LogAssert+0x15c9
clr.dll!LogHelp_LogAssert+0x6109
clr.dll!LogHelp_LogAssert+0x1d81
clr.dll!LogHelp_LogAssert+0x1cfc
clr.dll!LogHelp_LogAssert+0x1c3a
clr.dll!LogHelp_LogAssert+0x1dbb
clr.dll!LogHelp_LogAssert+0x6070
clr.dll!LogHelp_LogAssert+0x2bd8
clr.dll!LogHelp_LogAssert+0x2ab5
clr.dll!GetMetaDataInternalInterfaceFromPublic+0x6110f
KERNEL32.dll!BaseThreadInitThunk+0xd
ntdll.dll!RtlUserThreadStart+0x1d


SQLite.Interop.dll!sqlite3_enable_shared_cache+0x5aa0
SQLite.Interop.dll!sqlite3_backup_init+0xb561
SQLite.Interop.dll!sqlite3_backup_init+0x7f4b
SQLite.Interop.dll!sqlite3_backup_init+0x78db
SQLite.Interop.dll!sqlite3_expired+0x1c95
SQLite.Interop.dll!sqlite3_blob_open+0x270f
SQLite.Interop.dll!sqlite3_step+0x2fd
SQLite.Interop.dll!sqlite3_step+0x9c
[Native Frame: IL Method without Metadata]
[Managed to Unmanaged Transition]
System.Data.SQLite.dll!System.Data.SQLite.SQLite3.Step+0x2f3
System.Data.SQLite.dll!System.Data.SQLite.SQLiteDataReader.NextResult+0x1b7
System.Data.SQLite.dll!System.Data.SQLite.SQLiteDataReader..ctor+0x13c
System.Data.SQLite.dll!System.Data.SQLite.SQLiteCommand.ExecuteReader+0x3c
Duplicati.Library.Main.dll!Duplicati.Library.Main.Database.ExtensionMethods.ExecuteReader+0x17b
Duplicati.Library.Main.dll!<GetBlocklists>d__88.MoveNext+0x1a5
Duplicati.Library.Main.dll!<CreateIndexVolume>d__0.MoveNext+0x5f2
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!MoveNextRunner.Run+0x6c
mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction+0x61
mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations+0x3da
mscorlib.dll!System.Threading.Tasks.Task`1.TrySetResult+0x65
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.TrySetResult+0x14
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.SetResult+0xa
Duplicati.Library.Main.dll!<<DoRunOnMain>b__1>d.MoveNext+0x18e
mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start+0xb1
Duplicati.Library.Main.dll!<>c__DisplayClass5_0`1.<DoRunOnMain>b__1+0x93
Duplicati.Library.Main.dll!<Start>d__4.MoveNext+0x118
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!MoveNextRunner.Run+0x6c
mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction+0x61
mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations+0x3da
mscorlib.dll!System.Threading.Tasks.Task`1.TrySetResult+0x65
mscorlib.dll!System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.SetResult+0x9a
CoCoL.dll!<ReadAsync>d__29.MoveNext+0xc95
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!MoveNextRunner.Run+0x6c
mscorlib.dll!System.Threading.Tasks.AwaitTaskContinuation.RunOrScheduleAction+0x61
mscorlib.dll!System.Threading.Tasks.Task.FinishContinuations+0x3da
mscorlib.dll!System.Threading.Tasks.Task`1.TrySetResult+0x65
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.TrySetResult+0x14
mscorlib.dll!System.Threading.Tasks.TaskCompletionSource`1.SetResult+0xa
mscorlib.dll!System.Threading.ExecutionContext.RunInternal+0x160
mscorlib.dll!System.Threading.ExecutionContext.Run+0x14
mscorlib.dll!System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem+0x70
mscorlib.dll!System.Threading.ThreadPoolWorkQueue.Dispatch+0x14e
[Unmanaged to Managed Transition]
clr.dll!LogHelp_LogAssert+0x1013
clr.dll!LogHelp_LogAssert+0xeb6
clr.dll!LogHelp_LogAssert+0x15c9
clr.dll!LogHelp_LogAssert+0x6109
clr.dll!LogHelp_LogAssert+0x1d81
clr.dll!LogHelp_LogAssert+0x1cfc
clr.dll!LogHelp_LogAssert+0x1c3a
clr.dll!LogHelp_LogAssert+0x1dbb
clr.dll!LogHelp_LogAssert+0x6070
clr.dll!LogHelp_LogAssert+0x2bd8
clr.dll!LogHelp_LogAssert+0x2ab5
clr.dll!GetMetaDataInternalInterfaceFromPublic+0x6110f
KERNEL32.dll!BaseThreadInitThunk+0xd
ntdll.dll!RtlUserThreadStart+0x1d

**Comment by kimmerin** [comment]
Here is the ZIP-file containing the dumps. I had to remove half of the trace due to the size limit for the upload.
[Duplicati_Traces.zip](https://github.com/duplicati/duplicati/files/2006489/Duplicati_Traces.zip)

**Comment by Ruvetuve** [comment]
Iam about to backup TB's of data with duplicati. But this issue prevents me from doing so.

Is this issue known with the devs and what is the plan to resolve this? No clear statement and the issue is reported over a year ago.

As long as the backup is not interrupted it seems to avoid this issue correct?

**Comment by kenkendk** [comment]
@StevenKSanford The query you found to be blocking is a repair step.
When a remote file is missing, Duplicati will attempt to recreate it.
If the missing file is a `dblock` file, that means that actual file contents are missing.
Duplicati tries to reconstruct the `dblock` file by looking at your local files and then extracts the data from these files.

If you have a large backup this can take a long time and the database query is probably not optimized at all. Not sure what the best option is here, but the rebuilding of the `dblock` files is rarely possible, because the (meta-) data is no longer present (timestamps updated, etc).

Maybe a better solution is to default to not attempting to create the blocks as it is extremely slow, and unlikely to work, and instead suggest that the broken volume is purged.

@afalout Your query stops at a different place, and I think it is part of an integrity or cleanup check. However, that query is no longer present in Duplicati, so it should at least not stop on that issue again.

@instantlinux Can you provide me with a copy (privately) of a broken database? And the exact error message, then I can have a look.

**Comment by duplicatibot** [comment]
This issue has been mentioned on **Duplicati**. There might be relevant details there:

https://forum.duplicati.com/t/lost-the-database-still-have-local-files-and-access-to-remote-files-how-do-i-get-things-running-again/10072/16

**Comment by StevenKSanford** [comment]
I haven't seen this problem in quite a while. Assuming it has been fixed, I'll close this issue. Thanks to everyone for your help with this... Steve
